metricq/history_client.py

```diff
@@ -116,30 +117,36 @@ def __init__(self, proto: history_pb2.HistoryResponse, request_duration=None):
         count = len(proto.time_delta)
         if count == 0:
             self._mode = HistoryResponseType.EMPTY
-            assert (
-                len(proto.value_min) == 0
-                and len(proto.value_max) == 0
-                and len(proto.aggregate) == 0
-                and len(proto.value) == 0
-            ), "Inconsistent HistoryResponse message"
+            if (
+                len(proto.value_min) != 0
+                or len(proto.value_max) != 0
+                or len(proto.aggregate) != 0
+                or len(proto.value) != 0
+            ):
+                raise HistoryResponseError("time_delta empty, but values present")
 
         elif len(proto.aggregate) == count:
             self._mode = HistoryResponseType.AGGREGATES
-            assert len(proto.value) == 0, "Inconsistent HistoryResponse message"
+            if len(proto.value) != 0:
+                raise HistoryResponseError("values and aggregates present")
 
         elif len(proto.value) == count:
             self._mode = HistoryResponseType.VALUES
-            assert len(proto.aggregate) == 0, "Inconsistent HistoryResponse message"
+            if len(proto.aggregate) != 0:
+                raise HistoryResponseError("aggregates and values present")
 
         elif len(proto.value_avg) == count:
             self._mode = HistoryResponseType.LEGACY
-            assert len(proto.value_min) == count
-            assert len(proto.value_max) == count
-            assert len(proto.aggregate) == 0
-            assert len(proto.value) == 0
+            if (
+                len(proto.value_min) != count
+                or len(proto.value_max) != count
+                or len(proto.aggregate) != 0
+                or len(proto.value) != 0
+            ):
+                raise HistoryResponseError("inconsistent counts in legacy format")
 
         else:
-            raise ValueError("Inconsistent HistoryResponse message")
+            raise ValueError("no value count matches time_delta count")
```

Comment:
```suggestion
            raise HistoryResponseError("no value count matches time_delta count")
```

Reply:
We should also agree on a style for error messages. Usually they start with a capital letter.

Reply:
Do they? In our code, it is somewhat mixed and I think there are more with lower case.

Python itself:

```
SyntaxError: invalid syntax
IndexError: list index out of range
IndentationError: unexpected indent
```